src/integrationTest/java/uk/gov/hmcts/darts/cases/service/RetentionPolicyTest.java

```diff
@@ -0,0 +1,180 @@
+        List<CaseRetentionEntity> caseRetentionList = new ArrayList<>();
+        caseRetentionList.add(createPendingCaseRetention(minimalHearing.getCourtCase()));
+        minimalHearing.getCourtCase().setCaseRetentionEntities(caseRetentionList);
+
+        var hearingEntity = dartsDatabase.saveRetentionsForHearing(minimalHearing, caseRetentionList);
+        var singleCase = casesMapper.mapToSingleCase(hearingEntity.getCourtCase());
+
+        assertThat(singleCase.getRetentionPolicyApplied()).isNull();
+        assertThat(singleCase.getRetainUntilDateTime()).isNull();
+        assertThat(singleCase.getRetentionDateTimeApplied()).isNull();
+        assertThat(singleCase.getCaseClosedDateTime()).isEqualTo(hearingEntity.getCourtCase().getCaseClosedTimestamp());
+    }
+
+    @Test
+    void retrievePolicyDetails() {
+        Policy policybyPolicyName = Policy.getForPolicyName("DARTS Permanent Retention v3");
+        Policy policybyPolicyNumber = Policy.getForPolicyNumber("08170758956b0615");
+
+        assertThat(policybyPolicyName.name()).isEqualTo(String.valueOf(Policy.PERMANENT));
+        assertThat(policybyPolicyNumber.name()).isEqualTo(String.valueOf(Policy.STANDARD));
+
+    }
+
+    private static CaseRetentionEntity createCompleteCaseRetention1(CourtCaseEntity courtCase) {
+        CaseRetentionEntity caseRetention = new CaseRetentionEntity();
+        caseRetention.setRetentionPolicyType(createRetentionPolicyType());
+        caseRetention.setTotalSentence("20Y2M10D");
+        caseRetention.setRetainUntil(OffsetDateTime.parse("2019-08-31T15:20:30+08:00"));
+        caseRetention.setRetainUntilAppliedOn(OffsetDateTime.parse("2019-08-31T15:20:30+08:00"));
+        caseRetention.setCurrentState(String.valueOf(CaseStatus.COMPLETE));
+        caseRetention.setSubmitted(OffsetDateTime.parse("2019-08-31T15:20:30+08:00"));
+        caseRetention.setCourtCase(courtCase);
+        return caseRetention;
+    }
+
+    private static CaseRetentionEntity createCompleteCaseRetention2(CourtCaseEntity courtCase) {
+        CaseRetentionEntity caseRetention = new CaseRetentionEntity();
+        caseRetention.setRetentionPolicyType(createRetentionPolicyType());
+        caseRetention.setTotalSentence("20Y2M10X");
+        caseRetention.setRetainUntil(OffsetDateTime.parse("2020-08-31T15:20:30+08:00"));
+        caseRetention.setRetainUntilAppliedOn(OffsetDateTime.parse("2020-08-31T15:20:30+08:00"));
+        caseRetention.setCurrentState(String.valueOf(CaseStatus.COMPLETE));
+        caseRetention.setSubmitted(OffsetDateTime.parse("2020-08-31T15:20:30+08:00"));
+        caseRetention.setCourtCase(courtCase);
+        return caseRetention;
+    }
+
+    private static CaseRetentionEntity createPendingCaseRetention(CourtCaseEntity courtCase) {
+        CaseRetentionEntity caseRetention = new CaseRetentionEntity();
+        caseRetention.setRetentionPolicyType(createRetentionPolicyType());
+        caseRetention.setTotalSentence("20Y2M10Z");
```

Comment:
This format is wrong, needs to have a D on the end, not a Z.

Reply:
During testing to distinguish the data I updated the last character. I have reverted this back to D